### ISSUE TITLE
Improve management of fractional numeric values in Airbyte Glue integration

### ISSUE THREAD

**Issue by blarghmatey** [mitodl/ol-data-platform#708]
# Description/Context
Airbyte has a representation of data types that all source connectors map their own types into, and which the destination connectors need to map from into their own types ([reference](https://docs.airbyte.com/understanding-airbyte/supported-data-types/)). Numeric values that have fractional components are represented as a `number`. There are a handful of ways that this can be mapped into supported types in our Glue table definitions, each of which have their own tradeoffs.

We can map the numeric values to the compatible types available in the [Hive language reference](https://cwiki.apache.org/confluence/display/hive/languagemanual+types#LanguageManualTypes-NumericTypes):
- `float` - 4 byte floating point number
- `double` - 8 byte floating point number
- `decimal` - (this is what we currently use) - up to a 38 digit number with fixed precision (maximum digits) and scale (number of digits to the right of the decimal)

`float` and `double` types are flexible, but don't have any guarantees of accuracy in mathematical operations (problematic in financial contexts). `decimal` is precise and reliable for financial cases, but requires foreknowledge of how many digits to the right of the decimal are necessary.

We need to expose a configurable method of specifying the `type` of numeric value to use in the table definition, and for the `decimal` case we need to be able to specify the `scale` of the value so that we don't truncate the fractional portion of the numeric values.

# Plan/Design
- [ ] Add a way to specify which type to use for fractional numerics (`double` or `decimal`)
- [ ] For `decimal` types add a means of specifying the `scale` of the type to be created in the table
- [ ] The default behavior should be to create fractional numeric types as `decimal(38,2)`, but allowing the scale (the second parameter) to be specified in the connector configuration and passed through to the table schema generation.

**Comment by blarghmatey** [comment]
This is a useful reference to confirm our thinking around preferring the `decimal` type https://data.solita.fi/to-float-or-not-to-float-choosing-correct-numeric-data-type-in-snowflake/

**Comment by quazi-h** [comment]
Made changes to the Airbyte S3 Glue destination connector to add a configuration setting to specify numeric type and scale (decimal only) for fractional values.
Created a new branch https://github.com/mitodl/airbyte/tree/709-fractional-numeric-options-s3-glue from the stringify option changes to leverage the MetastoreFormatConfig we created to handle these parameters.
Created an upstream PR to get these changes merged to the official Airbyte code https://github.com/airbytehq/airbyte/pull/26517/files#diff-38a9968e3b671a7139138a90221d68cc94233a2382b068bbe9b23983feec7506
Worked with Tobias to get get these changes deployed to QA for testing and point all of our QA connections to our Airbyte fork.
Once these changes are merged in our fork and we have updated our Prod connections to pull from our Airbyte fork, this work will be done.
